.spectral.js

```diff
@@ -67,6 +67,7 @@ module.exports = {
       files: [
         "web-api/openapi/back_office.yaml#/components/schemas/DocumentCandidatesRequest",
         "web-api/openapi/back_office.yaml#/components/schemas/DocumentCandidatesResponse",
+        "web-api/openapi/schemas/document.yml#/DocumentPropertyArrayString",
```

Comment:
does anybody has an idea how to silence this lint, such that it doesn't produce the following error?
```
/__w/xayn_discovery_engine/xayn_discovery_engine/web-api/openapi/schemas/document.yml
 20:29  warning  ibm-array-attributes  Array schemas should define a numeric 'minItems' field  DocumentPropertyArrayString
 20:29  warning  ibm-array-attributes  Array schemas should define a numeric 'maxItems' field  DocumentPropertyArrayString
```
i also tried it with `"web-api/openapi/front_office.yaml#/components/schemas/FilterCompare/additionalProperties/properties/$in",` but to no effect 🤷

Reply:
I think we do want to define `minItems` & `maxItems` here :thinking: 

For `minItems` we want to be clear if it's 0+ or 1+.

For `maxItems` we can't just accept arbitrary long arrays AFIK, especially if we want to hold the option open to not use elasticsearch in the future. The really hard question is what to chooses for `maxItems`?

Reply:
> For minItems we want to be clear if it's 0+ or 1+.

one of the previous changes allowed empty arrays as valid, so we have to go with 0. but then `minItems: 0` is redundant, because without `minItems` this always holds according to the openapi docs.

> For maxItems we can't just accept arbitrary long arrays AFIK, especially if we want to hold the option open to not use elasticsearch in the future. The really hard question is what to chooses for maxItems?

we limit the overall byte size of the properties, so this automatically limits how much a specific property can hold, hence we don't have to put a `maxItems` limit.

we could differentiate the types though, one for `[string]` properties and one for `$in` filter arguments and could put a limit on the latter. not sure if this is too much though. the question would still be what limit.

Reply:
i've set it to `minItems: 0` and `maxItems: 100` for now. even though the upper limit is a breaking change it should be way more than enough for now and we can also increase it in the future.

regarding the linter, we may be able to implement our custom resolver which doesn't only inline `$ref`s from the root file but also all other `$ref`s from the non-root files, then the overrides should apply (maybe, i didn't get a fully clear picture from their docs). this needs to be tried by someone who knows javascript though. or we need to look for another linter.

Reply:
I think we should make `$in` have a different limit as both have very different performance properties, I also would say it makes sense to set some limit for both, through potential quite high limits.

I'm not sure if 100 is too small of an limit, we know of at least one use-case which wants a higher limit but probably so high that it might outstrip any limit.